### PR TITLE
Feat js t2003

### PR DESCRIPTION
**Pull request by sganeshkumaar**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -9,6 +9,9 @@ let cityNames
 /* array of cities to be displayed as cards */
```
Comment (lakshmipooja1510) [inline src/index.js:9]: Check line 6, 15, 51, 110, 119 for spelling mistake or missing space between 2 words

File: src/index.js [hunk 2]
```diff
@@ -173,16 +194,25 @@ function updateTopClock (dateAndTime) {
...
+  document.getElementById('factor-humidity').innerHTML = factors[2] + ' %'
+  document.getElementById('factor-precipitate').innerHTML = factors[3] + ' %'
+}
+
+/**
+ * @param {string} city name of the city
+ * @returns {Array} all four factors
+ */
+function getAndCalculateFactors (city) {
   const tempIncelsius = parseInt(coreData[city].temperature.slice(0, -2))
```
Comment (lakshmipooja1510) [inline src/index.js:210]: Make the variable in camel case

File: src/index.js [hunk 3]
```diff
@@ -9,6 +9,9 @@ let cityNames
 /* array of cities to be displayed as cards */
 let displayedCards
```
Comment (lakshmipooja1510) [inline src/index.js:10]: image
The time is not synced properly in top and middle section. Do check

File: src/index.js [hunk 4]
```diff
@@ -9,6 +9,9 @@ let cityNames
...
 
+/* array of cities to be displayed as tiles */
+let displayedTiles
+
 /* id for the clock setinterval in topsection */
 let timerId
```
Comment (lakshmipooja1510) [inline src/index.js:17]: The sorting in the section 3 is only in increasing order and u should do the same for ascending and make the arrow point accordingly

File: src/index.js [hunk 5]
```diff
@@ -24,8 +27,6 @@ async function loadFunction () {
```
Comment (lakshmipooja1510) [inline src/index.js:27]: What happens if both temp and continent is selected

Comment (sganeshkumaar) [reply src/index.js:27]: sorry i cant see the image

File: src/index.js [hunk 6]
```diff
@@ -500,29 +539,29 @@ function sortCityCards (filteredCities, preference) {
...
  */
 function runCityDates (cityNames) {
-  for (const i in cityNames) {
-    cityDates.set(cityNames[i], new Date(coreData[cityNames[i]].dateAndTime))
+  for (const city of cityNames) {
+    cityDates.set(city, new Date(liveDateAndTimeOfCity(city)))
   }
 
-  setInterval(updateCardAndTileTime, 60000)
+  setInterval(updateCardAndTileTime, 59900)
```
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js:546]: i thought the process would take some ms to execute . so 100 ms for the execution of code in interval

[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js:546]: if i give correctly 1000 ms in Update Top clock set interval , the clock seems to fall behind live time

File: src/index.js (hunk 1 above)
Comment (sganeshkumaar) [reply src/index.js:9]: all spelling mistakes resolved

File: src/index.js (hunk 2 above)
Comment (sganeshkumaar) [reply src/index.js]: changed to camel case

File: src/index.js (hunk 3 above)
Comment (sganeshkumaar) [reply src/index.js:10]: Clocks in all three sections are synced

File: src/index.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/index.js]: changed to 1000 in both set intervals

File: src/index.js (hunk 4 above)
Comment (sganeshkumaar) [reply src/index.js:17]: added ascending and descending order

**Comment by lakshmipooja1510** [comment]
Move the css files to a folder called styles inside src

**Comment by sganeshkumaar** [comment]
> Move the css files to a folder called styles inside src

css files added to a new folder called styles

File: src/generateTop.js [hunk 7]
```diff
@@ -0,0 +1,192 @@
+import { globalVariables } from './index.js'
```
[the remark this answers is not in the document]
Comment (sganeshkumaar) [reply src/generateTop.js]: that i intentionally placed to know if the time is running. i forgot to change the interval to 1000

**Pull request merged**